libs/evm2wasm/evm2wast.c

```diff
@@ -355,7 +355,7 @@ static inline void add_metering(char **wast, char **segment)
 		return;
 	}
 	sprintf(*wast, "%s (call $useGas (i64.const %ld)) %s", *wast, gas_count, *segment);
-	free(*segment);
+	*segment = "";
```

Comment:
This will put a literal into a pointer which is freed later on and throws away the pointer which needs to be freed (memory leak).